cpp/validate_harvested_records.cc

```diff
@@ -234,28 +287,31 @@ const std::map<std::string, std::string> EQUIVALENT_TAGS_MAP{
 void AnalyseNewJournalRecord(const MARC::Record &record, const bool first_record,
                              const GeneralInfo &general_info, JournalInfo * const journal_info)
 {
-    std::unordered_set<std::string> seen_tags;
+    std::unordered_set<std::string> seen_tags_and_subfield_codes;
     MARC::Tag last_tag;
     for (const auto &field : record) {
         auto current_tag(field.getTag());
         if (current_tag == last_tag)
             continue;
 
-        seen_tags.emplace(current_tag.toString());
+        const RecordType record_type(record.isReviewArticle() ? REVIEW : REGULAR_ARTICLE);
+        for (const auto &subfield : field.getSubfields()) {
+            if (general_info.find(current_tag.toString(), subfield.code_, record_type) != general_info.end())
+                continue;
 
-        if (general_info.find(current_tag.toString()) != general_info.end())
-            continue;
+            seen_tags_and_subfield_codes.emplace(current_tag.toString() + std::string(1, subfield.code_));
 
-        if (first_record)
-            journal_info->addField(current_tag.toString(), ALWAYS);
-        else if (journal_info->find(current_tag.toString()) == journal_info->end())
-            journal_info->addField(current_tag.toString(), SOMETIMES);
+            if (first_record)
+                journal_info->addField(current_tag.toString(), subfield.code_, ALWAYS, record_type);
```

Comment:
I don't think this is enough, here are some scenarios: 
- If we have any GeneralInfo/regular_article/700/SOMETIMES, we do not want a JournalInfo/700a/ALWAYS to be automatically added (no matter if theJournalInfo is regular_article or review)
- If we have a GeneralInfo/review/700/SOMETIMES, a JournalInfo/regular_article/700a/ALWAYS should be added (if there is not GeneralInfo/regular_article/700 or 700a present)